### ISSUE TITLE
Filter errors

### ISSUE THREAD

**Issue by alpascal** [joinbox/infect#30]
I've found another filter error: when a single substance is added to a substance class, the logical operator is AND instead of OR, resulting in the situation seen in the screen shot.
![capture](https://cloud.githubusercontent.com/assets/1735457/7278002/41baf36e-e913-11e4-905e-8000443f77d7.PNG)

**Comment by fxstr** [comment]
@alpascal Not sure if I understand the problem. If I filter for two substance classes, the filter seems to be applied correctly (OR): 

![image](https://cloud.githubusercontent.com/assets/1211144/7681976/a8228fb6-fd74-11e4-9d62-5ae699dbc6dd.png)

If I add a name filter (anti biotic «Cotri…»), the filtering seems to be correct again:
`(Fluroi…  OR Makroli…) AND Cotri…`

![image](https://cloud.githubusercontent.com/assets/1211144/7681988/be2f319c-fd74-11e4-9e44-5a725cee195c.png)

Or is that behaviour different from what you'd expect?

**Comment by alpascal** [comment]
You're right, I'm not sure if it's really a bug, as this may be just an issue of the logic operator used. IMO it should show Cotrimoxazole(Name), Fluorchinolones(Substance class) and Macrolids(Substance class) all together if all are selected, as it does when multiple substance classes are selected.
This would be the most useful view if one were to compare the susceptibility of one substance with a(nother) substance class.

**Comment by alpascal** [comment]
Went over this again: problem is that when user filters for a compound and a substance class, the operator used is AND. To elaborate, if I intend to show only the compound "ciprofloxacin" and then the whole substanceclass "makrolide" to compare the two, then I would filter for the compound "ciprofloxacin" (is substanceclass "fluorchinolone") and the substanceclass "makrolide". With the operator AND used at the moment, this correctly shows nothing, because there is no compound called "ciprofloxacin" belonging to the substanceclass "makrolide".
To solve this issue, the operator should be changed to "OR" for filtering all compounds and substanceclasses together.

**Comment by fxstr** [comment]
Planned for 1.2 release
